frontend/src/metabase/actions/containers/ActionCreatorModal/ActionCreatorModal.tsx

```diff
@@ -11,22 +20,60 @@ interface OwnProps {
     actionId?: string;
   };
   onClose: () => void;
+  onChangeLocation: (location: LocationDescriptor) => void;
 }
 
-interface ModelLoaderProps {
+interface EntityLoaderProps {
+  action?: WritebackAction;
   model: Card;
+  loading?: boolean;
 }
 
-type ActionCreatorModalProps = OwnProps & ModelLoaderProps;
+interface DispatchProps {
+  setErrorPage: (error: AppErrorDescriptor) => void;
+  onChangeLocation: (location: LocationDescriptor) => void;
+}
+
+type ActionCreatorModalProps = OwnProps & EntityLoaderProps & DispatchProps;
+
+const mapDispatchToProps = {
+  setErrorPage,
+  onChangeLocation: replace,
+};
 
 function ActionCreatorModal({
+  action,
   model,
   params,
+  loading,
   onClose,
+  setErrorPage,
+  onChangeLocation,
 }: ActionCreatorModalProps) {
-  const { slug, actionId } = params;
-  const modelId = Urls.extractEntityId(slug);
+  const actionId = Urls.extractEntityId(params.actionId);
+  const modelId = Urls.extractEntityId(params.slug);
   const databaseId = model.database_id || model.dataset_query.database;
+
+  useEffect(() => {
+    if (loading === false) {
```

Comment:
Yeah, I was surprised about that as well :/ IMO, we should have `loading: true` instead of `loading: undefiend` there. Or maybe replace them with a `status` enum like `react-query` does (could be `"idle" | "loading" | "done" | "error"` or something like that)